### DISCUSSION TITLE
visual studio 2022 redundancy error

### DISCUSSION THREAD

**Discussion by pedoc** [community/community#14156] [Copilot Conversations]
I don't think the intermediate files generated by copilot should appear in visual studio.

![image](https://user-images.githubusercontent.com/6604230/161413144-f567019b-0684-49ac-8218-2a1e0c993d97.png)

**Comment by kzu** [comment]
The "error" is an IntelliSense one only, it will not prevent compilation, and can prevented from appearing entirely if you select `Build Only` from the dropdown next to the errors filter icon. 

It will also automatically go away as soon as you close the Copilot toolwindow. The underlying issue is that to provide coloring on fragments of C#, we use temporary files and a miscellaneous files Roslyn workspace. The snippets aren't (of course) entirely valid full-C# files and in most cases the Roslyn analyzers should ignore them, but it seems you came across a case where it didn't. Any chance you can provide the full file context that triggered this?

Thanks!

**Reply by pedoc** [reply]
I haven't found steps to stably reproduce this issue. I'll keep tracking this bug though

PS:
OS:win11 22h2 22581.200
VS:2022 17.1.3
Github Copilot:1.10.0.3

**Reply by ctorx** [reply]
This happens to me as well.  

What's most frustrating about this is if you have the "Build + Intellisense" option selected in the Error List pane, you will often see a bunch of Co-Pilot errors but the actual error causing compilation failure won't be displayed. 

VS and Extension Info:
---------------------------------------------------------------------------
Microsoft Visual Studio Community 2022
Version 17.1.0
VisualStudio.17.Release/17.1.0+32210.238
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Community

.NET Core Debugging with WSL   1.0
.NET Core Debugging with WSL

ASP.NET and Web Tools 2019   17.1.358.51495
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   17.1.358.51495
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   17.1.358.51495
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   17.1.358.51495
Azure Functions and Web Jobs Tools

C# Tools   4.1.0-5.22109.6+0c82c4114a4e4b8b723b915eee3b13261db6717f
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GitHub Copilot   1.12.0.0 (v1.12.0.0@0660f7705)
GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.

GitHub Copilot Agent   1.12.5517 (v1.12.0)

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.161+abc97ecc7d.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio Tools for Containers   1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Node.js Tools   1.5.40105.1 Commit Hash:1822ee94b29c6cf748a19825f14cc26d30b0b871
Adds support for developing and debugging Node.js apps in Visual Studio

NuGet Package Manager   6.1.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.0.0.2206201+62a2c1d6162f828801565a7ca26d9d48b810a05b
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.0.62201.12090
Microsoft SQL Server Data Tools

TypeScript Tools   17.0.1229.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.1.0-5.22109.6+0c82c4114a4e4b8b723b915eee3b13261db6717f
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.21610.4+07b5673e4f2fa7630e78abe37f16b372353a7242
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Inline Suggestions   2.2.1930.216
Ghost text API for Visual Studio inline suggestions

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

**Reply by kzu** [reply]
Could you provide some repro steps? I think this could only happen if you have the Open Copilot (toolwindow) open showing multiple suggestions...

**Comment by ctorx** [comment]
I don't have specific repo steps.

I installed Co-Pilot and began using it in an existing moderately sized
solution.

I had recently installed VS2022 (one month earlier), and the Error pane was
defaulted to Errors+Intellisense.

In my case, I tend to see this happen when I've refactored something
(changed the name of a class or a method), and the razor page or MVC view
is still referencing the former name.

The compilation failure is being triggered by the view/page (.cshtml) but
all of the errors in the list are for Co-Pilot, and thus, you have no idea
what the problem is.

I must have restarted VS and rebooted 30 times, thinking it was a cache
issue before I came upon this thread and learned that I could disable the
intellisense on the error pane, which makes the problem go away.

**Reply by kzu** [reply]
> all of the errors in the list are for Co-Pilot, and thus, you have no idea what the problem is.

Could you please clarify the scenario there? Copilot will not introduce any errors/warnings unless suggestions include partial code (that doesn't compile for whatever reason) and ONLY if you open them in the Open Copilot toolwindow. Is that the case or are you seeing the errors appear in some other circumstance?

The next release will contain additional checks to ensure the temporary files used by copilot are properly deleted as well as the error list cleared as soon as you accept (or close) a suggestion from that toolwindow.
